devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,680 @@
+	}
+}
+
+private channelNumber(String deviceNetworkId) {
+	deviceNetworkId.split(":")[-1] as Integer
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+private void setState(value, endpoint = null) {
+	def map = [
+		encap(zwave.basicV1.basicSet(value: value), endpoint),
+		encap(zwave.switchBinaryV1.switchBinaryGet(), endpoint),
+	]
+	sendHubCommand(map, 500)
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+def childOn(childId) {
+	setState(0xFF, channelNumber(childId))
+}
+
+def childOff(childId) {
+	setState(0, channelNumber(childId))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			map.name = "energy"
+			map.value = cmd.scaledMeterValue
+			map.unit = "kWh"
+			sendEvent(map)
+		} else if (cmd.scale == 2) {
+			map.name = "power"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "W"
+			sendEvent(map)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def numberOfButtons = state.numberOfButtons
+	def state
+	def buttonN
+	switch (cmd.keyAttributes as Integer) {
+		case 0:
+			state = "pushed"
+			buttonN = cmd.sceneNumber
+			break
+		case 1: 
+			state = "up"
+			buttonN = cmd.sceneNumber
+			break
+		case 2: 
+			state = "held"
+			buttonN = cmd.sceneNumber
+			break
+	}
+	if (buttonN) {
+		def buttonId = buttonN + numberOfButtons * 2
+		def child = childDevices?.find {channelNumber(it.deviceNetworkId) == buttonId }
+		child?.sendEvent([name: "button", value: state, data: [buttonNumber: 1], isStateChange: true])
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def numberOfButtons = state.numberOfButtons
+	def endPoint = cmd.sourceEndPoint.toInteger()
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
+	def value = encapsulatedCommand.value
+	def childDevice = childDevices?.find {channelNumber(it.deviceNetworkId) == endPoint }
+	def corrRelCons = 0
+	def corRelParam = 11 + endPoint - numberOfButtons
+	if (endPoint in numberOfButtons..(2*numberOfButtons)) {
+		def param = parameterMap().find( {it.paramNum == corRelParam } ).name
+		def paramState = state."$param"
+		if (paramState) {
+			corrRelCons = paramState.value
+		}
+	}
+	if (childDevice) {
+		childDevice.sendEvent(name: "switch", value: value ? "on" : "off")
+		if (value) {
+			sendEvent(name: "switch", value: "on")
+			childDevice.sendEvent(name: "power", value: corrRelCons, unit: "W")
+		} else {
+			childDevice.sendEvent(name: "power", value: 0, unit: "W")
+			if (!childDevices.any { it.currentValue("switch") == "on" }) {
+				sendEvent(name: "switch", value: "off")
+			}
+		}
+	}
```

Comment:
I'm still gonna insist on using `zwaveEvent()` here.

Reply:
@Sarkis008 what about this one ?

Reply:
@KKlimczukS I used zwaveEvent() 

https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/0783e698e9fa5c72b524d7e851adce75201c304c/devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy#L265